bots: Replace whitelist with GitHub /contributors call

Gundersanne opened the pull request.

allisonkarlitskaya commented on bots/task/github.py:
```diff
@@ -325,6 +299,17 @@ def commits(self, branch='master', since=None):
+    page = 1
+    count = 100
+    while count == 100:
+        data = api.get("/contributors?page={0}&per_page={1}".format(page, count)) or [];
```
I'd rather we limit this to the cockpit contributors team, rather than the full contributors list.  That's too many people, considering we're running arbitrary code with access to the cockpit bots' github token (which can do a lot of stuff).

martinpitt commented:
FTR, from IRC discussion, and validating with my own and with cockpituous' token: 

 - First do https://api.github.com/orgs/cockpit-project/teams and get the team ID
 - Then do https://api.github.com/teams/12345/members to get the members

Preferrable to move `.whitelist()` as a method into the GitHub class, as it needs the token. This requires updating all callers, but there's only a handful of them.

Gundersanne commented:
GitHub().whitelist() might raise an exception when a team with a certain name is not found. I thought that would be better than silently failing.

Squashed and added Fixes/Closes.

martinpitt commented on bots/task/github.py:
```diff
@@ -325,6 +300,25 @@ def commits(self, branch='master', since=None):
+    def teamIdFromName(self, name):
+        try:
+            return next(team for team in self.get("/orgs/cockpit-project/teams") or [] if team.get('name') == TEAM_CONTRIBUTORS).get('id')
+        except StopIteration:
```
This is a really odd way of writing this search. How about this:

```py
for team in self.get("/orgs/cockpit-project/teams", []):
    if team.get("name") == TEAM_CONTRIBUTORS:
         return team["id"]
else:
    raise KeyError(".. not found")
```

Please also use consistent quoting and cause a `KeyError` if "id" is not present (the above does that).

mvollmer commented on bots/task/github.py:
```diff
@@ -325,6 +300,26 @@ def commits(self, branch='master', since=None):
+
+    def teamIdFromName(self, name):
+        for team in self.get("/orgs/cockpit-project/teams") or []:
+            if team.get("name") == TEAM_CONTRIBUTORS:
```
This should be `name`, the parameter, I think.

mvollmer commented on bots/task/github.py:
```diff
@@ -325,6 +300,26 @@ def commits(self, branch='master', since=None):
+
+    def teamIdFromName(self, name):
+        for team in self.get("/orgs/cockpit-project/teams") or []:
+            if team.get("name") == TEAM_CONTRIBUTORS:
```
This should be name, the parameter, I think, and not TEAM_CONTRIBUTORS.

Gundersanne replied:
Woops, thanks for catching that. My fault for just copying.

allisonkarlitskaya commented on bots/task/github.py:
```diff
@@ -60,38 +59,14 @@
 
-# the user name is accepted if it's found in either list
-WHITELIST = os.path.join(BASE, "bots", "whitelist")
-WHITELIST_LOCAL = "~/.config/github-whitelist"
```
Do we want to keep this, perhaps (the local override)?

martinpitt replied:
We don't use that on the bots, and that's the only relevant place. So I'm in favor of dropping it.

Gundersanne replied:
`WHITELIST` should be removed together with the whitelist file in the project I think.

 If we'd keep `WHITELIST_LOCAL` I'd avoid doing the override silently.

martinpitt commented:
With this patch, `api.whitelist()` says

    {'suomiy', 'jniederm', 'thomasmckay', 'cockpituous', 'mareklibra', 'allisonkarlitskaya', 'martinpitt', 'garrett', 'larskarlitski', 'andreasn', 'Gundersanne', 'stefwalter', 'petervo', 'aweiteka', 'sgratch', 'jscotka'}

which looks correct.
